packages/lit-dev-content/site/tutorials/content/reactivity/03.md

```diff
@@ -0,0 +1,30 @@
+Setting a reactive property or calling `requestUpdate()` triggers a reactive update cycle. The update cycle is _asynchronous_, so that updating several properties in the course of a single event handler, for example, only triggers a single update.
+
+At the beginning of an update cycle, Lit calls the `shouldUpdate()` method, passing in a map of changed properties.
+
+You can override `shouldUpdate()` to determine whether the component needs to update itself.
```

Comment:
I was thinking the opposite – why is duration a reactive prop anyway rather than a public class member unless you're using the attribute to pre-select the default option.

+1 to steve's suggestion

Reply:
@sorvell The animation happens when the user clicks a button. Triggering an update when the animation speed changes doesn't show anything, because there's no animation running. I did think about something like a key or unique ID--but a) this didn't really make sense to me, unless "Bob"s record never changes, and b) then I'd need to .... I don't know, pick from a list of users with associated data objects? Not sure how to make this into a demo.

I also considered something like an object with a "timestamp" field, where you only rerender if the timestamp is newer than the previous one.  But again, it's a fairly complicated demo setup.

@e111077 Yeah, I had the same though, and my only justification is that you could set it from an attribute. It is not much of a justification, but there you have it.

Reply:
Seems fine as is. We could also mention that in `shouldUpdate` you get the complete list of changes and can decide to filter them as appropriate. In this case, for example, you could do `return !(props.size === 1 && props.has('duration'))`... feels more natural for this case.

Reply:
Makes sense. Oh, should probably update this to `changedProperties` which is what I used in another step & is more self-documenting.

Reply:
Fixed.